### PR TITLE
[DOCS] Document empty first line support for msearch API

### PR DESCRIPTION
**Pull request by jrodewig**
Adds an 8.0 breaking change for PR #41011

### Preview
https://elasticsearch_78284.docs-preview.app.elstc.co/guide/en/elastic-stack/master/elasticsearch-breaking-changes.html#_search_changes

### CODE REVIEW

File: docs/reference/migration/migrate_8_0/search.asciidoc [hunk 1]
```diff
@@ -6,6 +6,25 @@
...
+====
+*Details* +
+The multi search API now supports an empty first line when you provide the
+request body as a text file, such as when using curl's `--data-binary` flag. The
+API parses an empty first line as empty action metadata.
+
+The API no longer supports text files that contain:
+
+* An empty first line followed by a line containing only `{}`.
+* An empty first line followed by another empty line.
```
Comment (jrodewig) [inline docs/reference/migration/migrate_8_0/search.asciidoc:21]: I originally included some example snippets here, but it looks like we strip out leading whitespace. ☹️ 
I couldn't figure out a good workaround. Let me know if this text is enough without a visual aid.

File: docs/reference/migration/migrate_8_0/search.asciidoc [hunk 2]
```diff
@@ -6,6 +6,25 @@
...
+====
+*Details* +
+The multi search API now parses an empty first line as empty action metadata
+when you provide the request body as a text file, such as when using curl's
+`--data-binary` flag.
+
+The API no longer supports text files that contain:
+
+* An empty first line followed by a line containing only `{}`.
+* An empty first line followed by another empty line.
```
Comment (jrodewig) [inline docs/reference/migration/migrate_8_0/search.asciidoc:21]: I originally included some example snippets here, but it looks like we strip out leading whitespace in snippets. ☹️ 
I couldn't figure out a good workaround. Let me know if this text is enough without a visual aid.

**Comment by jrodewig** [comment]
I originally included some example snippets here, but it looks like we strip out leading whitespace in snippets. I couldn't figure out a good workaround. Let me know if this text is enough without a visual aid.

**Review by cbuescher — APPROVED** [review]
LGTM, I think the text is clear enough.

**Comment by jrodewig** [comment]
Thanks @cbuescher.

**Pull request merged**